File: TOMIMSV4/Entity/Transacciones/OrdenCompra/OC_Imagen/clsBeTrans_oc_imagen.vb
Public Class clsBeTrans_oc_imagen
    Implements ICloneable
    Implements IDisposable

    Public Property IdOrdenCompraImg() As Integer = 0
    Public Property IdOrdenCompraEnc() As Integer = 0
    Public Property Orden() As Integer = 0
    Public Property Imagen() As Byte()
    Public Property Descripcion() As String = ""

    Sub New()
    End Sub

    Sub New(ByRef IdOrdenCompraImg As Integer, ByVal IdOrdenCompraEnc As Integer, ByVal Orden As Integer, ByVal Imagen As Byte(), ByVal descripcion As String)
        Me.IdOrdenCompraImg = IdOrdenCompraImg
        Me.IdOrdenCompraEnc = IdOrdenCompraEnc
        Me.Orden = Orden
        Me.Imagen = Imagen
        Me.Descripcion = descripcion
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Trans_oc/clsBeTrans_oc_imagen.cs
namespace WMS.EntityCore.Trans_oc
{
    public class clsBeTrans_oc_imagen : ICloneable, IDisposable
    {
        public int IdOrdenCompraImg { get; set; } = 0;
        public int IdOrdenCompraEnc { get; set; } = 0;
        public int Orden { get; set; } = 0;
        public byte[] Imagen { get; set; } = Array.Empty<byte>();
        public string Descripcion { get; set; } = string.Empty;
        public bool IsNew { get; set; } = false;

        public clsBeTrans_oc_imagen() { }

        public clsBeTrans_oc_imagen(int idOrdenCompraImg, int idOrdenCompraEnc, int orden, byte[] imagen, string descripcion)
        {
            IdOrdenCompraImg = idOrdenCompraImg;
            IdOrdenCompraEnc = idOrdenCompraEnc;
            Orden = orden;
            Imagen = imagen ?? Array.Empty<byte>(); // Ensure no null assignment  
            Descripcion = descripcion;
        }

        public object Clone()
        {
            return MemberwiseClone();
        }

        public void Dispose()
        {
            // Liberación manual de recursos si fuese necesario  
            Imagen = Array.Empty<byte>(); // Replace null assignment with an empty array  
            GC.SuppressFinalize(this);
        }
    }
}
